### DISCUSSION TITLE
Is it feasible to reuse the frontend from a community-provided backend port for a different target platform (e.g. JVM)?

### DISCUSSION THREAD

**Discussion by preslavrachev** [pocketbase/pocketbase#2190] [Q&A]
Hi @ganigeorgiev,

PocketBase is fantastic! I use it for some personal Go projects and love it. Unfortunately, for some Java projects I've been participating with other teams on, it's not always an easy option to host it as a separate service. I've been sort of exercising an idea to port parts of the backend in Java, so that JVM folks can include it as a JAR, and get the frontend "for free". While I work on frontend stuff all the time, my design skills are far from the best, so I'd highly appreciate working with something as polished as PB.

How feasible is that from your point of you? I assume that things are still being changed a lot, but at some point, you will end up at a backend API that is more or less backwards-compatible right? If there is enough interest from the JVM comunity, we could run the port as part of the PocketBase organization. I do not mind at all.

Cheers from a fellow Bulgarian, living in Germany 👋

P:S. Linking to https://github.com/pocketbase/pocketbase/discussions/1620, since it is on a related tangent.

**Comment by ganigeorgiev** [comment]
If your backend implements the same [public web APIs](https://pocketbase.io/docs/api-records/), then I guess it shouldn't be very difficult to embed the [Admin UI SPA](https://github.com/pocketbase/pocketbase/tree/master/ui) in your project.

But please note that, at least until we manage to complete most of the [planned tasks](https://github.com/orgs/pocketbase/projects/2), I can't make any BC promises. I try to avoid breaking changes in the public web APIs, but sometimes they are unavoidable due to some not very well thought earlier design choices.

There are also plans to refactor the Admin UI in the future to allow loading external plugins/modules as part of https://github.com/pocketbase/pocketbase/issues/898.
